### ISSUE TITLE
Is it possible to run this on the CPU alone?

### ISSUE THREAD

**Issue by takuman9** [google-research/frame-interpolation#35]
Nice to meet you, thank you for providing great software. Is it possible to run this on the CPU alone?

I would be happy if it could work in a poor edge environment where cuDNN doesn't work.

**Comment by fitsumreda** [comment]
Thanks for your interest in our work.
The test codes should already run fine as is. The training and benchmark eval scripts can also run on a cpu, you simply append '--cpu'

**Comment by fitsumreda** [comment]
You don't need to add for this.

**Comment by tralala87** [comment]
Ok, for what then? I am interested in using CPU as well, but your answer did not answer the raised issue from OP.

**Comment by fitsumreda** [comment]
You do not need to add a `--cpu` flag in the `python -m frame_interpolation.eval.interpolator_test ...` command. If a GPU is available, it will run on it, otherwise it should run fine on a cpu.

**Comment by tralala87** [comment]
Many thanks.

**Comment by takuman9** [comment]
@fitsumreda 
Thank you for your answer.

I tested it in a CPU-only environment and an environment with a GPU.
I saw the following message at run time, so I knew it would switch automatically.

CPU 
`tensorflow/core/framework/cpu_allocator_impl.cc:80] Allocation of 741343232 exceeds 10% of free system memory.`

GPU 
`tensorflow/core/common_runtime/bfc_allocator.cc:272] Allocator (GPU_0_bfc) ran out of memory trying to allocate 2.03GiB with freed_by_count=0. The caller indicates that this is not a 
failure, but may mean that there could be performance gains if more memory were available.`

Thank you very much.
